### PR TITLE
Tijlleenders/issue345

### LINKED ISSUE #345: fix i236
**Describe the bug**
When a Step is >8 hours it doesn't get split into 1hour Steps. 
The [business rule](https://github.com/tijlleenders/ZinZen-scheduler/issues/318#issuecomment-1597069630) was not clear enough. It should be:  
```
if 0 < step.duration <= 8 hours
  - make it one step
else:
  - split task into many steps with 1h duration each
```

**To Reproduce**
bug_236 in rust_tests

**Expected behavior**
See bug description.

### PR DESCRIPTION
**Pull request by tijlleenders**
Resolves #345

### CODE REVIEW

**Comment by tijlleenders** [comment]
@moaz-mokhtar  
The primary issue is resolved - but now there are a number of other test case outputs affected.  
For each one I'll have to manually check the expectation - and correct expectation or file a regression bug.  
This will take a while - but just FYI that I'm working on this.

**Comment by tijlleenders** [comment]
image

**Comment by tijlleenders** [comment]
@moaz-mokhtar  
Taking a different approach so this can get merged faster.  
- Ignoring all the test cases without status changed (OK=>OK or FAILED=>FAILED) - even if actual output changed.
- Checking OK=>FAILED
  - split_2 - corrected expectation ; now OK again
- Checking FAILED=>OK
  - bug_236 as expected ; fixing this was the objective of this PR
  - impossible_2 ; corrected expectation ; now still OK
  - issue_284_no_repeat_large_dur_and_timespan ; was a formatting issue - logged #348

File: tests/jsons/bug-236/actual_output.json
[the remark this answers is not in the document]
Comment (moaz-mokhtar) [reply tests/jsons/bug-236/actual_output.json:1]: So I got confirmation that process is as below:
```
if 0 < step.duration <= 8 hours
  - make it one step
else:
  - split all into many steps with 1h duration
```

**Review by moaz-mokhtar — APPROVED** [review]

**Pull request merged**